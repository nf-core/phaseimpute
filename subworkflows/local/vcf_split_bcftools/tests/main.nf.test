nextflow_workflow {

    name "Test Subworkflow VCF_SPLIT_BCFTOOLS"
    script "../main.nf"

    config "./nextflow.config"

    workflow "VCF_SPLIT_BCFTOOLS"

    tag "subworkflows"
    tag "subworkflows_local"
    tag "subworkflows/vcf_split_bcftools"
    tag "vcf_split_bcftools"

    tag "bcftools"
    tag "bcftools/split"

    test("Split vcf file") {
        setup {
            run("BCFTOOLS_MERGE") {
                script "../../../../modules/nf-core/bcftools/merge/main.nf"
                process {
                """
                input[0] = Channel.of(
                    [
                        [id: "allSamples"],
                        [file(params.pipelines_testdata_base_path + "hum_data/individuals/NA12878/NA12878.s_imputed.bcf", checkIfExist:true),
                        file(params.pipelines_testdata_base_path + "hum_data/individuals/NA19401/NA19401.s_imputed.bcf", checkIfExist:true),
                        file(params.pipelines_testdata_base_path + "hum_data/individuals/NA20359/NA20359.s_imputed.bcf", checkIfExist:true)],
                        [file(params.pipelines_testdata_base_path + "hum_data/individuals/NA12878/NA12878.s_imputed.bcf.csi", checkIfExist:true),
                        file(params.pipelines_testdata_base_path + "hum_data/individuals/NA19401/NA19401.s_imputed.bcf.csi", checkIfExist:true),
                        file(params.pipelines_testdata_base_path + "hum_data/individuals/NA20359/NA20359.s_imputed.bcf.csi", checkIfExist:true)],
                        []
                    ]
                )
                input[1] = Channel.of([
                    [id: "GRCh38"],
                    file(params.pipelines_testdata_base_path + "hum_data/reference_genome/GRCh38.s.fa.gz", checkIfExist:true),
                    file(params.pipelines_testdata_base_path + "hum_data/reference_genome/GRCh38.s.fa.gz.fai", checkIfExist:true),
                ])
                """
                }
            }
        }
        when {
            params {
                max_cpus   = 2
                max_memory = '2.GB'
            }
            workflow {
                """
                input[0] = BCFTOOLS_MERGE.out.merged_variants.join(BCFTOOLS_MERGE.out.tbi)
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(workflow.out).match() },
                { assert snapshot(workflow.out.vcf_tbi.collect{
                    path(it[1]).vcf.summary
                    }).match("Phasing content")
                }
            )
        }
    }
}
