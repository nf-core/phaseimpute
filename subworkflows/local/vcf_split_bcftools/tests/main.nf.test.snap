{
    "Split vcf file": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "NA12878"
                        },
                        "NA12878.vcf.gz:md5,af16cd340d33ffecc15dd6062b243232",
                        "NA12878.vcf.gz.tbi:md5,31c12acb72d301cc69478c58b513d779"
                    ],
                    [
                        {
                            "id": "NA19401"
                        },
                        "NA19401.vcf.gz:md5,09e2e2a30260310fe7db55edcf36b1c9",
                        "NA19401.vcf.gz.tbi:md5,8f93c5e7d94089d9b0313f1d7d7ba40d"
                    ],
                    [
                        {
                            "id": "NA20359"
                        },
                        "NA20359.vcf.gz:md5,56cba6b61bcc14c42d92bf3708ebf621",
                        "NA20359.vcf.gz.tbi:md5,6bf326636fb91d7ffffad543763c5be1"
                    ]
                ],
                "1": [
                    "versions.yml:md5,6c3351d97e3a99f7a7a3231fc49f92e2"
                ],
                "vcf_tbi": [
                    [
                        {
                            "id": "NA12878"
                        },
                        "NA12878.vcf.gz:md5,af16cd340d33ffecc15dd6062b243232",
                        "NA12878.vcf.gz.tbi:md5,31c12acb72d301cc69478c58b513d779"
                    ],
                    [
                        {
                            "id": "NA19401"
                        },
                        "NA19401.vcf.gz:md5,09e2e2a30260310fe7db55edcf36b1c9",
                        "NA19401.vcf.gz.tbi:md5,8f93c5e7d94089d9b0313f1d7d7ba40d"
                    ],
                    [
                        {
                            "id": "NA20359"
                        },
                        "NA20359.vcf.gz:md5,56cba6b61bcc14c42d92bf3708ebf621",
                        "NA20359.vcf.gz.tbi:md5,6bf326636fb91d7ffffad543763c5be1"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,6c3351d97e3a99f7a7a3231fc49f92e2"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-07-22T15:45:19.206308102"
    },
    "Phasing content": {
        "content": [
            [
                "VcfFile [chromosomes=[chr21, chr22], sampleCount=1, variantCount=1739, phased=true, phasedAutodetect=true]",
                "VcfFile [chromosomes=[chr21, chr22], sampleCount=1, variantCount=1739, phased=true, phasedAutodetect=true]",
                "VcfFile [chromosomes=[chr21, chr22], sampleCount=1, variantCount=1739, phased=true, phasedAutodetect=true]"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-07-18T17:13:33.060063904"
    }
}